cmd/fleetctl/preview.go

```diff
@@ -207,6 +207,13 @@ Use the stop and reset subcommands to manage the server and dependencies once st
 				return errors.New("Expected 1 active enroll secret")
 			}
 
+			// disable anonymous analytics collection for preview
+			if err := client.ApplyAppConfig(map[string]map[string]bool{
```

Comment:
We should start thinking about how we'll write tests for preview, because it's one of the worse things that can fail for us as it's what we tell customers to try.

Reply:
Yeah I think it'd pretty much need to be an e2e test, i.e. run `fleetctl preview` as an external command, and make sanity checks by calling it e.g. to get the app config and queries and stuff like that. As pure unit tests with mocked stuff, it looks pretty hard to create a relevant test (other than just checking "it called this, and then that", etc. which would be too tied to the internal implementation and otherwise not super useful IMO).

Reply:
Agreed. Potentially, with the smoke tests we run by hand it'll be enough.

Reply:
:+1: I'll take a shot at it after the second fleetctl preview issue/PR.